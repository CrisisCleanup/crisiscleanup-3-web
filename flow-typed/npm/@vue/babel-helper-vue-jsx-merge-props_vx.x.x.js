// flow-typed signature: d7a23c53f4091ef2bf1df8afa5eb43c4
// flow-typed version: <<STUB>>/@vue/babel-helper-vue-jsx-merge-props_v^1.0.0/flow_v0.131.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/babel-helper-vue-jsx-merge-props'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/babel-helper-vue-jsx-merge-props' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/babel-helper-vue-jsx-merge-props/dist/helper' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/babel-helper-vue-jsx-merge-props/dist/helper.js' {
  declare module.exports: $Exports<'@vue/babel-helper-vue-jsx-merge-props/dist/helper'>;
}
