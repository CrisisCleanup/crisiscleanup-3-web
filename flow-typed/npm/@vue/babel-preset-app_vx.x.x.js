// flow-typed signature: 6ba91a92845d9a556a5bcdcc4f033fed
// flow-typed version: <<STUB>>/@vue/babel-preset-app_v^4.3.1/flow_v0.129.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/babel-preset-app'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/babel-preset-app' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/babel-preset-app/polyfillsPlugin' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/babel-preset-app/index' {
  declare module.exports: $Exports<'@vue/babel-preset-app'>;
}
declare module '@vue/babel-preset-app/index.js' {
  declare module.exports: $Exports<'@vue/babel-preset-app'>;
}
declare module '@vue/babel-preset-app/polyfillsPlugin.js' {
  declare module.exports: $Exports<'@vue/babel-preset-app/polyfillsPlugin'>;
}
