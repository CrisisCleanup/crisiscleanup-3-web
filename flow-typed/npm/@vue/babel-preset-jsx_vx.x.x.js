// flow-typed signature: eeecaeff29e4a3a6c63c4fe7e16a0eb3
// flow-typed version: <<STUB>>/@vue/babel-preset-jsx_v^1.1.2/flow_v0.131.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/babel-preset-jsx'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/babel-preset-jsx' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/babel-preset-jsx/dist/plugin.cjs' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/babel-preset-jsx/dist/plugin.cjs.js' {
  declare module.exports: $Exports<'@vue/babel-preset-jsx/dist/plugin.cjs'>;
}
