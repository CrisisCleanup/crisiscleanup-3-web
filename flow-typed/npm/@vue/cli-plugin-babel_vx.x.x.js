// flow-typed signature: 4cf1a98e80b05c0a167e013d64bc9645
// flow-typed version: <<STUB>>/@vue/cli-plugin-babel_v^4.4.6/flow_v0.131.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/cli-plugin-babel'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/cli-plugin-babel' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/customConfig.input' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/customConfig.output' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/default.input' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/default.output' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/doubleQuote.input' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/doubleQuote.output' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/require.input' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/require.output' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/templateLiteral.input' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/templateLiteral.output' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/usePluginPreset' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/generator' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/migrator' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/preset' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/customConfig.input.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/customConfig.input'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/customConfig.output.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/customConfig.output'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/default.input.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/default.input'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/default.output.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/default.output'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/doubleQuote.input.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/doubleQuote.input'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/doubleQuote.output.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/doubleQuote.output'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/require.input.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/require.input'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/require.output.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/require.output'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/templateLiteral.input.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/templateLiteral.input'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/templateLiteral.output.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/templateLiteral.output'>;
}
declare module '@vue/cli-plugin-babel/codemods/usePluginPreset.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/usePluginPreset'>;
}
declare module '@vue/cli-plugin-babel/generator.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/generator'>;
}
declare module '@vue/cli-plugin-babel/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel'>;
}
declare module '@vue/cli-plugin-babel/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel'>;
}
declare module '@vue/cli-plugin-babel/migrator/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/migrator'>;
}
declare module '@vue/cli-plugin-babel/migrator/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/migrator'>;
}
declare module '@vue/cli-plugin-babel/preset.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/preset'>;
}
