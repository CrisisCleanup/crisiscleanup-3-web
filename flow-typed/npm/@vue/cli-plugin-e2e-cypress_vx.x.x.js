// flow-typed signature: 122252e4d027555936ded476aff636b4
// flow-typed version: <<STUB>>/@vue/cli-plugin-e2e-cypress_v^4.4.6/flow_v0.129.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/cli-plugin-e2e-cypress'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/cli-plugin-e2e-cypress' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/cli-plugin-e2e-cypress/generator' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/_eslintrc' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/specs/test' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/commands' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/ui' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/cli-plugin-e2e-cypress/generator/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/_eslintrc.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/_eslintrc'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/specs/test.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/specs/test'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/commands.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/commands'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support'>;
}
declare module '@vue/cli-plugin-e2e-cypress/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress'>;
}
declare module '@vue/cli-plugin-e2e-cypress/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress'>;
}
declare module '@vue/cli-plugin-e2e-cypress/ui.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/ui'>;
}
