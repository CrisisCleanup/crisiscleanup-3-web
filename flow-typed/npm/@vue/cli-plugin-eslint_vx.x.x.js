// flow-typed signature: 190fd4925231c8aa9cbfe00ac40be4f1
// flow-typed version: <<STUB>>/@vue/cli-plugin-eslint_v^4.4.6/flow_v0.129.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/cli-plugin-eslint'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/cli-plugin-eslint' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/cli-plugin-eslint/eslintDeps' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-eslint/eslintOptions' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-eslint/generator' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-eslint/lint' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-eslint/migrator' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-eslint/prompts' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-eslint/ui/configDescriptor' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-eslint/ui' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-eslint/ui/taskDescriptor' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/cli-plugin-eslint/eslintDeps.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/eslintDeps'>;
}
declare module '@vue/cli-plugin-eslint/eslintOptions.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/eslintOptions'>;
}
declare module '@vue/cli-plugin-eslint/generator/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/generator'>;
}
declare module '@vue/cli-plugin-eslint/generator/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/generator'>;
}
declare module '@vue/cli-plugin-eslint/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint'>;
}
declare module '@vue/cli-plugin-eslint/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint'>;
}
declare module '@vue/cli-plugin-eslint/lint.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/lint'>;
}
declare module '@vue/cli-plugin-eslint/migrator/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/migrator'>;
}
declare module '@vue/cli-plugin-eslint/migrator/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/migrator'>;
}
declare module '@vue/cli-plugin-eslint/prompts.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/prompts'>;
}
declare module '@vue/cli-plugin-eslint/ui/configDescriptor.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/ui/configDescriptor'>;
}
declare module '@vue/cli-plugin-eslint/ui/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/ui'>;
}
declare module '@vue/cli-plugin-eslint/ui/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/ui'>;
}
declare module '@vue/cli-plugin-eslint/ui/taskDescriptor.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/ui/taskDescriptor'>;
}
