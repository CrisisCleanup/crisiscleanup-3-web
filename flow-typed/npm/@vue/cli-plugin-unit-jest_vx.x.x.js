// flow-typed signature: 345a830bc132c538d84f7a5ce68b6d33
// flow-typed version: <<STUB>>/@vue/cli-plugin-unit-jest_v^4.4.6/flow_v0.131.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/cli-plugin-unit-jest'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/cli-plugin-unit-jest' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/cli-plugin-unit-jest/generator' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-unit-jest/generator/template/tests/unit/example.spec' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-unit-jest/jest-preset' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-unit-jest/presets/default/jest-preset' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-unit-jest/presets/no-babel/esmoduleTransformer' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-unit-jest/presets/no-babel/jest-preset' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-unit-jest/presets/typescript-and-babel/jest-preset' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-unit-jest/presets/typescript/jest-preset' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-unit-jest/ui' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/cli-plugin-unit-jest/generator/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-unit-jest/generator'>;
}
declare module '@vue/cli-plugin-unit-jest/generator/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-unit-jest/generator'>;
}
declare module '@vue/cli-plugin-unit-jest/generator/template/tests/unit/example.spec.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-unit-jest/generator/template/tests/unit/example.spec'>;
}
declare module '@vue/cli-plugin-unit-jest/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-unit-jest'>;
}
declare module '@vue/cli-plugin-unit-jest/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-unit-jest'>;
}
declare module '@vue/cli-plugin-unit-jest/jest-preset.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-unit-jest/jest-preset'>;
}
declare module '@vue/cli-plugin-unit-jest/presets/default/jest-preset.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-unit-jest/presets/default/jest-preset'>;
}
declare module '@vue/cli-plugin-unit-jest/presets/no-babel/esmoduleTransformer.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-unit-jest/presets/no-babel/esmoduleTransformer'>;
}
declare module '@vue/cli-plugin-unit-jest/presets/no-babel/jest-preset.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-unit-jest/presets/no-babel/jest-preset'>;
}
declare module '@vue/cli-plugin-unit-jest/presets/typescript-and-babel/jest-preset.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-unit-jest/presets/typescript-and-babel/jest-preset'>;
}
declare module '@vue/cli-plugin-unit-jest/presets/typescript/jest-preset.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-unit-jest/presets/typescript/jest-preset'>;
}
declare module '@vue/cli-plugin-unit-jest/ui.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-unit-jest/ui'>;
}
