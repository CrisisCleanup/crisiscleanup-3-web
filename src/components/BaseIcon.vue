<template>
  <div class="flex items-center base-icon">
    <img :class="styles" :src="iconMap[type]" :alt="alt" :title="alt" />
    <slot></slot>
  </div>
</template>

<script>
/* eslint-disable global-require */
const iconMap = {
  dashboard: require('@/assets/icons/dashboard.svg'),
  cases: require('@/assets/icons/cases.svg'),
  print: require('@/assets/icons/print-big.svg'),
  phone: require('@/assets/icons/phone.svg'),
  share: require('@/assets/icons/share-big.svg'),
  download: require('@/assets/icons/download.svg'),
  active: require('@/assets/icons/active.svg'),
  edit: require('@/assets/icons/edit.svg'),
  flag: require('@/assets/icons/flag.svg'),
  'flag-filled': require('@/assets/icons/flag-filled.svg'),
  search: require('@/assets/icons/search.svg'),
  table: require('@/assets/icons/table.svg'),
  map: require('@/assets/icons/notactive.svg'),
  info: require('@/assets/icons/inform.svg'),
  trash: require('@/assets/icons/delete.svg'),
  organization: require('@/assets/icons/my-organization.svg'),
  history: require('@/assets/icons/history.svg'),
  cancel: require('@/assets/icons/big.svg'),
  help: require('@/assets/icons/help.svg'),
  up: require('@/assets/icons/up.svg'),
  down: require('@/assets/icons/down.svg'),
  updown: require('@/assets/icons/updown.svg'),
  'go-case': require('@/assets/icons/replace-case.svg'),
  'map-buffer': require('@/assets/icons/map-buffer.svg'),
  'map-circle': require('@/assets/icons/map-circle.svg'),
  'map-poly': require('@/assets/icons/map-poly.svg'),
  'map-rect': require('@/assets/icons/map-rect.svg'),
  'map-sweep': require('@/assets/icons/map-sweep.svg'),
  'map-undo': require('@/assets/icons/map-undo.svg'),
  'map-redo': require('@/assets/icons/map-redo.svg'),
};
/* eslint-enable global-require */

export default {
  name: 'BaseIcon',
  props: {
    type: {
      type: String,
      default: '',
    },
    alt: {
      type: String,
      default: '',
    },
    size: {
      type: String,
      default: '',
    },
  },
  data() {
    return {
      iconMap,
      styles: {
        'ccu-icon': true,
        'cursor-pointer': true,
        large: this.size === 'large',
        medium: this.size === 'medium',
        small: this.size === 'small',
        xs: this.size === 'xs',
        xl: this.size === 'xl',
      },
    };
  },
};
</script>

<style scoped>
.ccu-icon {
  height: 30px;
  width: 30px;
}
.ccu-icon.medium {
  height: 20px;
  width: 20px;
}
.ccu-icon.xl {
  height: 35px;
  width: 35px;
}
.ccu-icon.small {
  height: 16px;
  width: 16px;
}
.ccu-icon.xs {
  height: 10px;
  width: 10px;
}
.filter-gray {
  filter: invert(84%) sepia(0%) saturate(30%) hue-rotate(209deg)
    brightness(107%) contrast(90%);
}
.filter-yellow {
  filter: invert(92%) sepia(21%) saturate(3995%) hue-rotate(346deg)
    brightness(98%) contrast(106%);
}
</style>
