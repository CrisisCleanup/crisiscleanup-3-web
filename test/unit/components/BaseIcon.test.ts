/**
 *
 * Tests for BaseIcon
 *
 * Components
 */

import { describe, expect, it, vi } from 'vitest';
import { mount } from '@vue/test-utils';
import BaseIcon from '@/components/BaseIcon.vue';

describe('BaseIcon', () => {
  it('should not log any errors', () => {
    const spy = vi.spyOn(global.console, 'error');
    mount(BaseIcon, {
      mocks: {
        $t: (key: string) => key,
      },
      stubs: {
        'base-button': true,
      },
    });
    expect(spy).not.toHaveBeenCalled();
  });

  it('should render correctly and match snapshot', () => {
    const testProps = {
      mocks: {
        $t: (key: string) => key,
      },
      props: { size: 'small', type: 'dashboard' },
    };
    const wrapper = mount(BaseIcon, testProps);
    expect(wrapper.element).toMatchSnapshot();
  });
});
