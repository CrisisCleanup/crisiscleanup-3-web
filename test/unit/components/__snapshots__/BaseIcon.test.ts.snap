// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseIcon > should render correctly and match snapshot 1`] = `
<base-button
  action="() => {
      }"
  alt="button"
  class="flex items-center base-icon js-button"
  data-v-92245bab=""
  title="button"
>
  <img
    alt="button"
    class="ccu-icon cursor-pointer small"
    data-v-92245bab=""
    src="/src/assets/icons/dashboard.svg"
  />
  
  
</base-button>
`;
